client/src/components/Dataset/services.ts

```diff
@@ -0,0 +1,77 @@
+import { fetcher, paths } from "schema";
+import { safePath } from "utils/redirect";
+
+const _getDatasets = fetcher.path("/api/datasets").method("get").create();
+type getDatasetsApiOptions = paths["/api/datasets"]["get"]["parameters"]["query"];
+type getDatasetsQuery = Pick<paths["/api/datasets"]["get"]["parameters"]["query"], "limit" | "offset">;
```

Comment:
```suggestion
import type { FetchArgType } from "openapi-typescript-fetch";
import { fetcher, paths } from "schema";
import { safePath } from "utils/redirect";

const _getDatasets = fetcher.path("/api/datasets").method("get").create();
type getDatasetsApiOptions = FetchArgType<typeof _getDatasets>;
type getDatasetsQuery = Pick<getDatasetsApiOptions, "limit" | "offset">;
```

Make types reliant on each other to avoid repeat definitions. This leads to stronger and more maintainable types.

Reply:
Ah, that's much cleaner, thank you!

Reply:
Hmm, getting never for the types in that case ...
<img width="777" alt="Screenshot 2022-12-06 at 11 45 15" src="https://user-images.githubusercontent.com/6804901/205890477-041c13cd-5f9f-462e-b98d-529c099c688a.png">

Reply:
That's strange. The tests seem to be failing too, but it works as expected on my end.

![image](https://user-images.githubusercontent.com/44241786/205894627-d3933091-c158-437c-b7ae-6c818e0ef67a.png)

Reply:
I've rebased the branch against dev, so maybe it has to do with the updated tsconfig ?

Reply:
huh, now it works for me too, maybe I messed up the conflicts ...

Reply:
Still works on the latest commit from this PR 
![image](https://user-images.githubusercontent.com/44241786/205895647-f3ca7a5e-6f9a-4fb6-af81-fcbb7b801120.png)

Reply:
The test failure might have been unrelated to this change, as it seems to have occurred in files not connected to this file. Try adding it again